ggsdk/sdk.py

```diff
@@ -1,24 +1,31 @@
+"""GraphGrid SDK"""
 import typing
 
 from ggcore.config import SdkBootstrapConfig
 from ggcore.core import SdkCore
 
 
 class GraphGridSdk:
-    """Initializes the SDK Core for SDK calls. Exposes those SDK calls as methods."""
+    """Initializes the SDK Core for SDK calls. Exposes those SDK calls as
+    methods. """
     _core: SdkCore
     _config: SdkBootstrapConfig
 
-    # todo init so they can just pass in a config map instead of individual params like above?
+    # todo init so they can just pass in a config map instead of individual
+    #  params like above?
     def __init__(self, access_key, secret_access_key, url_base="localhost"):
-        self._config = SdkBootstrapConfig(url_base, access_key, secret_access_key)
+        self._config = SdkBootstrapConfig(url_base, access_key,
+                                          secret_access_key)
         self._setup_core()
 
     def _setup_core(self):
         self._core = SdkCore(self._config)
 
     def test_api(self):
+        """Call test api"""
         return self._core.test_api()
 
-    def save_dataset(self, data_generator: typing.Generator, dataset_id: str = None, overwrite=False):
+    def save_dataset(self, data_generator: typing.Generator,
+                     dataset_id: str = None, overwrite=False):
+        """Call save dataset api"""
```

Comment:
It may be helpful to include parameter descriptions as well since this is the user facing function,
e.g. [here](https://bitbucket.org/graphgrid/nlp-model-training/src/e325126a9beac2bf5ea5b5837dbc1faa9ff0d0a7/src/retrieve_dependencies.py#lines-25:39)

Which then can be further parsed out within IDEs, like below:

![doc_string_example](https://user-images.githubusercontent.com/46761773/158630839-d03a98df-496a-44b6-9dbd-d295285ce77b.png)
)

as well as situations where users call `help(save_dataset)` in python, e.g.
```python
help(retrieve_dependencies)
Help on function retrieve_dependencies in module src.retrieve_dependencies:
retrieve_dependencies(local_train_path: str, local_eval_path: str, model_save_path: str, cache_path: str, dataset_bucket: str, endpoint_url: str, datasets: dict, datasets_is_str: bool)
    Retrieve all the dependencies for the model to be trained
    
    :param local_train_path: Local path to training files
    :param local_eval_path: Local path to evaluation files
    :param model_save_path: Local path to previously trained model
    :param cache_path: Local path to caching directory
    :param dataset_bucket: Bucket where training and evaluation files reside
    :param endpoint_url: Cloud environment endpoint url
    :param datasets: Dictionary containing all remote file paths for datasets
    :param datasets_is_str: Flag for whether datasets is assumed to be a str```

Reply:
And then this information can also be directly re-used on the doc site